src/Robot.elm

```diff
@@ -0,0 +1,162 @@
+module Robot exposing (..)
+
+import Browser
+import Browser.Events as Events exposing (..)
+import Html exposing (button, div, h2, text)
+import Html.Attributes as Attributes exposing (style)
+import Html.Events as Events
+import Json.Decode
+import Keyboard exposing (..)
+import Keyboard.Arrows exposing (..)
+import Time exposing (..)
+
+
+type alias RobotData =
+    { xPos : Int
+    , yPos : Int
+    , toggleSpeed : Bool
+    }
+
+
+setRobotData : Int -> Int -> Bool -> RobotData
```

Comment:
This function is pure - doesn't have any side effect (i.e. mutating some state).
It only creates new stuff.
So don't name it "set" - the name implies mutating inner state.
Remember, Elm isn't Java.